Samples/Client/SwiftClient/SwiftClient/VideoStreamViewController.swift

```diff
@@ -0,0 +1,640 @@
+        if (isFingerDown) {
+            if recognizer.numberOfTouches == 1 {
+                // location of first finger
+                let tappedPoint = recognizer.location(ofTouch: 0, in: self.view)
+                
+                let dx = tappedPoint.x - fingerDownX
+                let dy = tappedPoint.y - fingerDownY
+                
+                let dpitch = 0.005 * dy
+                let dheading = 0.005 * dx
+                
+                navHeading = downHeading - dheading
+                navPitch = downPitch + dpitch
+                
+                let locTransform =  matMultiply(a: matRotateY(rad: navHeading), b: matRotateZ(rad: navPitch))
+                navTransform = matMultiply(a: matTranslate(v: navLocation), b: locTransform)
+                
+                sendTransform()
+            }
+            
+            if recognizer.numberOfTouches == 2 {
+                // location of second finger
+                let tappedPoint = recognizer.location(ofTouch: 1, in: self.view)
+                
+                let dy = tappedPoint.y - fingerDownY
+                
+                let dist = -dy
+                
+                navLocation[0] = downLocation[0] + dist * navTransform[0]
+                navLocation[1] = downLocation[1] + dist * navTransform[1]
+                navLocation[2] = downLocation[2] + dist * navTransform[2]
+                
+                navTransform[12] = navLocation[0]
+                navTransform[13] = navLocation[1]
+                navTransform[14] = navLocation[2]
+                
+                sendTransform()
+            }
+        }
+    }
+    
+    
+    func sendTransform() {
+        if inputChannel != nil {
+            var eye = [ navTransform[12], navTransform[13], navTransform[14]]
+            var lookat = [ navTransform[12] + navTransform[0],
+                           navTransform[13] + navTransform[1],
+                           navTransform[14] + navTransform[2]]
+            var up = [ navTransform[4], navTransform[5], navTransform[6]]
+            
+            let data = "\(eye[0]), \(eye[1]), \(eye[2]), \(lookat[0]), \(lookat[1]), \(lookat[2]), \(up[0]), \(up[1]), \(up[2])"
+            
+            let msg = [
+                "type" : "camera-transform-lookat",
+                "body" : data
+            ]
+            
+            let buffer = RTCDataBuffer(data: convertDictionaryToData(dict: msg)!, isBinary: false)
+            inputChannel.sendData(buffer)
+        }
+    }
+    
+    func handlePeerMessage(peerId: Int, data: Data) {
+        messageCounter += 1
+        let stringData = String(data: data, encoding: .utf8)!
+        print("Message from \(otherPeers[peerId]!): \(stringData)")
+        
+        var dataJson = [String: Any]()
+        do {
+            dataJson = try JSONSerialization.jsonObject(with: data) as! [String: Any]
+        } catch {
+            print("Error deserializing JSON: \(error)")
```

Comment:
The user will not see the print if there is an error, and this will catch only all of the errors possible when deserializing (just from the do block), so I think it is fine to leave as is but can change it if you'd like

Reply:
if this fails to parse we likely won't see any data (av, etc) exchanged, right? If there's an easy way to surface this error to the user (like a message: "sdp negotiation failed, please restart the app") that would be awesome - we do try and surface this stuff when we can in our other clients.